plugins/http_plugin/http_plugin.cpp

```diff
@@ -766,16 +269,32 @@ class http_plugin_impl : public std::enable_shared_from_this<http_plugin_impl> {
              "Additionaly acceptable values for the \"Host\" header of incoming HTTP requests, can be specified multiple times.  Includes http/s_server_address by default.")
             ("http-threads", bpo::value<uint16_t>()->default_value( my->thread_pool_size ),
              "Number of worker threads in http thread pool")
+            ("http-keep-alive", bpo::value<bool>()->default_value(true),
+             "If set to false, do not keep HTTP connections alive, even if client requests.")
             ;
    }
 
    void http_plugin::plugin_initialize(const variables_map& options) {
       try {
-         my->validate_host = options.at("http-validate-host").as<bool>();
+         my->plugin_state->validate_host = options.at("http-validate-host").as<bool>();
          if( options.count( "http-alias" )) {
             const auto& aliases = options["http-alias"].as<vector<string>>();
-            my->valid_hosts.insert(aliases.begin(), aliases.end());
+            my->plugin_state->valid_hosts.insert(aliases.begin(), aliases.end());
          }
+         
+         my->plugin_state->max_body_size = options.at( "max-body-size" ).as<uint32_t>();
+         verbose_http_errors = options.at( "verbose-http-errors" ).as<bool>();
+         my->plugin_state->verbose_http_errors = verbose_http_errors;
```

Comment:
I dislike `verbose_http_errors` being a static namespace level variable. I dislike even more storing both in the static and in plugin_state. Since it was a static before, just remove it from plugin_state, or if you are feeling ambitious remove the static and only have the one in plugin_state.

Reply:
Agreed, unfortunately it had to remain static because its used inside http_plugin::handle_exception which is a static function.  

The reason I added it to plugin_state was to avoid a circular dependency between the beast_http_XXX and http_plugin.   Might make more sense to move it common.hpp.

Reply:
? it is not used by beast_http_XXX.